bpfman-operator/apis/v1alpha1/cluster_fentry_program_types.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2024 The bpfman Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+package v1alpha1
+
+type ClFentryProgramInfo struct {
+	ClFentryLoadInfo `json:",inline"`
+
+	// links is an optional field and is a flag to indicate if the FEntry program
+	// should be attached. The attachment point for a FEntry program is a Linux
+	// kernel function. Unlike other eBPF program types, an FEntry program must be
+	// provided with the target function at load time. The links field is optional,
+	// but unlike other program types where it represents a list of attachment
+	// points, for FEntry programs it contains at most one entry that determines
+	// whether the program should be attached to the specified function. To attach
+	// the program, add an entry to links with mode set to Attach. To detach it,
+	// remove the entry from links.
+	// +optional
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=1
+	Links []ClFentryAttachInfo `json:"links,omitempty"`
```

Comment:
Just to remind myself, you wanted to keep this as a list to maintain consistency with other program types right?